apps/scene2pset/scene2pset.cc

```diff
@@ -43,8 +44,76 @@ struct AppSettings
     float min_valid_fraction = 0.0f;
     float scale_factor = 2.5f; /* "Radius" of MVS patch (usually 5x5). */
     std::vector<int> ids;
+    bool output_correspondence = false;
 };
 
+struct CorrespondenceViewMetadata
+{
+    unsigned int view_id; 
+    unsigned int width;  // After downsampling. 
+    unsigned int height; // After downsampling.
+    unsigned long first_idx;
+
+    CorrespondenceViewMetadata (unsigned int view_id, unsigned int width, unsigned int height, unsigned long first_idx):
```

Comment:
I'd remove the constructor here. It makes the code below a bit more verbose, but also more clear. And structs shouldn't have logic, even simple constructor logic.